### ISSUE TITLE
[Issue]: Lyrics are not refreshed even with "Replace all Metadata"

### ISSUE THREAD

**Issue by mihawk90** [jellyfin/jellyfin#12148]
### Please describe your bug

Due to formatting issues tracked in #11518 I had to re-format some of my timestamps. Unfortunately, I cannot get JF to pick up the changes.
I tried everything from simply scanning for updated files to Replacing all Metadata *including images*, and none of them update the extracted Lyrics.

### Reproduction Steps

1. Have some tracks with Lyrics already extracted
2. Change lyrics tags in the metadata
3. Try and of the various Refresh Metadata options

### Jellyfin Version

10.9.0

### if other:

10.9.6

### Environment

```markdown
- OS: TrueNAS Scale
- Linux Kernel: Linux truenas 6.6.29-production+truenas #1 SMP PREEMPT_DYNAMIC Wed May 29 15:05:57 UTC 2024 x86_64 GNU/Linux
- Virtualization: Docker/Kubernetes (TrueNAS official App)
- Clients: Browser
- Browser: Vivaldi
- FFmpeg Version: 6.0.1-Jellyfin
- Playback Method: -
- Hardware Acceleration: -
- GPU Model: -
- Plugins: -
- Reverse Proxy: -
- Base URL: -
- Networking: whatever Docker does by default 😅
- Storage: local
```

### Jellyfin logs

```shell
[2024-06-21 03:53:02.924 +00:00] [INF] [11] Emby.Server.Implementations.Session.SessionWebSocketListener: Sending ForceKeepAlive message to 1 inactive WebSockets.
[2024-06-21 03:53:38.924 +00:00] [INF] [53] Emby.Server.Implementations.Session.SessionWebSocketListener: Sending ForceKeepAlive message to 1 inactive WebSockets.
[2024-06-21 03:54:56.385 +00:00] [INF] [29] MediaBrowser.MediaEncoding.Encoder.MediaEncoder: Starting "/usr/lib/jellyfin-ffmpeg/ffprobe" with args "-analyzeduration 200M -probesize 1G -i file:\"/mnt/media/music/Podcasts/A State of Trance/ASOT/A State Of Trance Episode 945 (#ASOT945) [TOP 50 Of 2019 Special] – Armin van Buuren [FN0ENLG8O3s].ogg\" -threads 0 -v warning -print_format json -show_streams -show_format"
[2024-06-21 03:55:01.471 +00:00] [INF] [20] Jellyfin.Plugin.KodiSyncQueue.EntryPoints.LibrarySyncNotification: Started library sync
[2024-06-21 03:55:02.156 +00:00] [INF] [20] Jellyfin.Plugin.KodiSyncQueue.EntryPoints.LibrarySyncNotification: Library Sync: Updated 1 items
[2024-06-21 03:55:02.156 +00:00] [INF] [20] Jellyfin.Plugin.KodiSyncQueue.EntryPoints.LibrarySyncNotification: Finished library sync, taking "00:00:00.6844622"
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by joshuaavalon** [comment]
I faced the same problem. I use Mp3tag for my tag. Tracks that added before 10.9 have lyrics but after 10.9 did not.
I have also tried to refresh metadata on track, album and artist.

**Comment by mihawk90** [comment]
> Tracks that added before 10.9 have lyrics but after 10.9 did not.

That's a different issue then.
This bug is specifically about Lyrics already in JF's database (or files or whatever it uses) not being updated.

Also for the record: All my files were present before 10.9 but the extraction happened after 10.9 (obviously, since Lyrics support was added in 10.9). I did add new tracks in 10.9 too and their lyrics were also extracted just fine.

**Comment by joshuaavalon** [comment]
@mihawk90 Neverminded. I just double check and find out the one has lyric have an extra text plus embedded lyric. It seems like embedded lyric does not work yet.

**Comment by mihawk90** [comment]
> It seems like embedded lyric does not work yet.

All my lyrics are embedded. I have no separate Lyrics files whatsoever (I hate file clutter).

But I believe if both a separate file *and* embedded lyrics are present, the separate file takes precedence.

**Comment by joshuaavalon** [comment]
@mihawk90 No, I think you misunderstand me. I mean embedded does not work before or after 10.9 (they don't have extra text file). I though the track that added before 10.9 work was because there is a text file there. Other track that does not have text file does not work no matter what.

**Comment by mihawk90** [comment]
> I mean embedded does not work before or after 10.9

As I said, all my Lyrics are embedded, and they show in Jellyfin. So yes, generally speaking, they work.

Whatever issue you're having is separate from the issue in my original post.

**Comment by tuxpizza** [comment]
Has anyone figured a way around this? Force re-extraction or lyrics? Even after *removing an entire album*, force refreshing metadata, then adding it back, it *still* uses an old cached version of the lyrics instead of the updated embedded lyrics. Such a major oversight

**Comment by mihawk90** [comment]
> Has anyone figured a way around this?

You can remove the lrc file from JF's metadata directory manually and then refresh metadata (might have to do it twice, it seems #11960 is still around in some shape or form).

**Comment by mihawk90** [comment]
By the way if you try to do the above... it is a bit of manual work, but this is how I did it:

```sh
$ cd /path/to/album
$ ls | grep -E ".mp3|.ogg|.m4a|.flac" > list.txt # add or remove file extensions as required
$ sed -i "s/.mp3/.lrc/g; s/.ogg/.lrc/g; s/.m4a/.lrc/g; s/.flac/.lrc/g" list.txt # add or remove file extensions as required
$ metapath=/path/to/jellyfin/metadata # in docker this'll just be /config/metadata
$ while IFS= read file; do find $metapath -name "$file" -print -quit; done <./list.txt # only search and print the filepaths
$ while IFS= read file; do find $metapath -name "$file" -print -delete -quit; done <./list.txt # if you're feeling adventurous you can also do this directly
$ rm list.txt
```

After that, Refresh/Replace Metadata from the WebUI (possibly twice as noted above), and you *should* see the new lyrics.

Obviously if you just have single tracks that changed you can just do `find $path -name xyz.ext -print -delete -quit` manually but I had to do it to 250 tracks and I'm not typing that :P

**Comment by mihawk90** [comment]
Behaviour still unchanged.

---

Side note, for this:

> Obviously if you just have single tracks that changed you can just do `find $metapath -name xyz.ext -print -delete -quit` manually

I don't know which version introduced this but there is now also an "edit lyrics" option in the context menu which will also let you remove the LRC.

**Comment by mihawk90** [comment]
Still the same, although I have not tested on 10.11 RC1.

**Comment by ngoc-minh-do** [comment]
I facing this issue

**Comment by mihawk90** [comment]
As of 10.11.4, this is still the case. I'm aware .5 just dropped a couple hours ago, and I can't update right now, but the release notes don't list anything related anyway.

The previously mentioned workaround of "Edit Lyrics" > "Delete" > "Refresh Metadata" > "Scan for new and updated files" still works though.

**Comment by mihawk90** [comment]
Still the same on 10.11.8.

This issue has been on Triage for 2 years now, can we please tag this so I don't get pinged every 3 months when there hasn't been a substantial new release anyway <_<

**Comment by IRHM** [comment]
Just encountered this bug on 10.11.10. (it's still reproducible)

Edit: Not sure if another issue is warranted (because I think the same fix likely fixes this too), but it seems this bug is not only limited to lyrics, so the fix hopefully fixes all tags not being picked up in Jellyfin after the initial discovery of the file (not sure if every tag isn't being updated, but for example, genres also doesn't seem to be picked up when edited).
